src/rpc/rawtransaction.cpp

```diff
@@ -926,59 +931,73 @@ static RPCHelpMan testmempoolaccept()
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Array cannot contain more than " + ToString(MAX_PACKAGE_COUNT) + " transactions.");
     }
 
-    CMutableTransaction mtx;
-    if (!DecodeHexTx(mtx, request.params[0].get_array()[0].get_str())) {
-        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed. Make sure the tx has at least one input.");
-    }
-    CTransactionRef tx(MakeTransactionRef(std::move(mtx)));
-
+    const UniValue raw_transactions = request.params[0].get_array();
     const CFeeRate max_raw_tx_fee_rate = request.params[1].isNull() ?
                                              DEFAULT_MAX_RAW_TX_FEE_RATE :
                                              CFeeRate(AmountFromValue(request.params[1]));
 
-    CTxMemPool& mempool = EnsureMemPool(request.context);
-    int64_t virtual_size = GetVirtualTransactionSize(*tx);
-    CAmount max_raw_tx_fee = max_raw_tx_fee_rate.GetFee(virtual_size);
+    std::vector<CTransactionRef> txns;
+    for (const auto& rawtx : raw_transactions.getValues()) {
+        CMutableTransaction mtx;
+        if (!DecodeHexTx(mtx, rawtx.get_str())) {
+            throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed. Make sure the tx has at least one input.");
+        }
+        txns.emplace_back(MakeTransactionRef(std::move(mtx)));
+    }
 
-    UniValue result(UniValue::VARR);
-    UniValue result_0(UniValue::VOBJ);
-    result_0.pushKV("txid", tx->GetHash().GetHex());
-    result_0.pushKV("wtxid", tx->GetWitnessHash().GetHex());
-
-    const MempoolAcceptResult accept_result = WITH_LOCK(cs_main, return AcceptToMemoryPool(::ChainstateActive(), mempool, std::move(tx),
-                                                  false /* bypass_limits */, /* test_accept */ true));
-
-    // Only return the fee and vsize if the transaction would pass ATMP.
-    // These can be used to calculate the feerate.
-    if (accept_result.m_result_type == MempoolAcceptResult::ResultType::VALID) {
-        const CAmount fee = accept_result.m_base_fees.value();
-        // Check that fee does not exceed maximum fee
-        if (max_raw_tx_fee && fee > max_raw_tx_fee) {
-            result_0.pushKV("allowed", false);
-            result_0.pushKV("reject-reason", "max-fee-exceeded");
-        } else {
-            result_0.pushKV("allowed", true);
-            result_0.pushKV("vsize", virtual_size);
-            UniValue fees(UniValue::VOBJ);
-            fees.pushKV("base", ValueFromAmount(fee));
-            result_0.pushKV("fees", fees);
+    CTxMemPool& mempool = EnsureMemPool(request.context);
+    CChainState& chainstate = EnsureChainman(request.context).ActiveChainstate();
+    const PackageMempoolAcceptResult validation_result = txns.size() > 1
+        ? WITH_LOCK(cs_main, return ProcessNewPackage(chainstate, mempool, txns, /* test_accept */ true))
+        : WITH_LOCK(cs_main, return PackageMempoolAcceptResult(txns[0]->GetHash(),
```

Comment:
I think we should use ProcessNewPackage here -- if this is done to avoid breaking RPC change, then I'd rather a new RPC call that consistently uses one call or the other.

Otherwise how to test if a single tx as a package is accepted?

Reply:
Yeah, there's a breaking RPC change if they try to testmempoolaccept on a tx that does an RBF (since we don't allow that in packages). I'm not really sure why a user would want to test a single tx as a package?

Reply:
maybe add a forceaspackage param bool? Would be good to be able to test the single tx as a package code path to test your PR, and a user would want to the same behavior call independent.

Eventually I think all relay should be going through the package path so I like to minimize special casing :)

Reply:
Oh I wasn't thinking of it that way - why should all relay go through the package path?

For tests, it's definitely possible to call `ProcessNewPackage` with a package of 1 transaction, it's just not exposed through RPC. I have the fuzzer doing it (see top commit)